yt_dlp/extractor/vrt.py

```diff
@@ -1,82 +1,215 @@
-        client = attrs.get('data-client-code') or self._CLIENT_MAP[site]
-
-        title = strip_or_none(get_element_by_class(
-            'vrt-title', webpage) or self._html_search_meta(
-            ['og:title', 'twitter:title', 'name'], webpage))
-        description = self._html_search_meta(
-            ['og:description', 'twitter:description', 'description'], webpage)
-        if description == '…':
-            description = None
-        timestamp = unified_timestamp(self._html_search_meta(
-            'article:published_time', webpage))
+        display_id = self._match_id(url)
+
+        model_json = self._download_json(f'{url.strip("/")}.model.json', display_id,
+                                         'Downloading asset JSON', 'Unable to download asset JSON')
+        details = model_json.get('details')
+        actions = details.get('actions')
+        title = details.get('title')
+        episode_publication_id = actions[2].get('episodePublicationId')
+        episode_video_id = actions[2].get('episodeVideoId')
+        video_id = f'{episode_publication_id}${episode_video_id}'
+        description = details.get('description')
+        episode = details.get('data').get('episode')
+        display_id = episode.get('name')
+        timestamp = parse_iso8601(episode.get('onTime').get('raw'))
+        upload_date = strftime('%Y%m%d', gmtime(timestamp))
+        series_info = details.get('data')
+        series = series_info.get('program').get('title')
+        season = series_info.get('season').get('title').get('value')
+        season_number = series_info.get('season').get('title').get('raw')
+        season_id = series_info.get('season').get('id')
+        episode = series_info.get('episode').get('number').get('value')
+        episode_number = series_info.get('episode').get('number').get('raw')
+        episode_id = series_info.get('episode').get('id')
+
+        video_info = None
+        vrtnutoken = ""
+
+        if self._authenticated:
+            vrtnutoken = self._download_json('https://token.vrt.be/refreshtoken',
+                                             video_id, note='refreshtoken: Retrieve vrtnutoken',
+                                             errnote='refreshtoken failed')['vrtnutoken']
+
+        headers = self.geo_verification_headers()
+        headers.update({'Content-Type': 'application/json; charset=utf-8'})
+        vrtPlayerToken = self._download_json(
+            '%s/tokens' % self._REST_API_BASE_TOKEN, video_id,
+            'Downloading token', headers=headers, data=json.dumps({
+                'identityToken': vrtnutoken
+            }).encode('utf-8'))['vrtPlayerToken']
+
+        video_info = self._download_json(
+            '%s/media-items/%s' % (self._REST_API_BASE_VIDEO, video_id),
+            video_id, 'Downloading video JSON', query={
+                'vrtPlayerToken': vrtPlayerToken,
+                'client': 'vrtnu-web@PROD',
+            }, expected_status=400)
```

Comment:
can simplify this a bit
```suggestion
        vrtnutoken = self._download_json(
            'https://token.vrt.be/refreshtoken', video_id, note='Retrieving vrtnutoken',
            errnote='Token refresh failed')['vrtnutoken'] if self._authenticated else ''

        vrt_player_token = self._download_json(
            'https://media-services-public.vrt.be/vualto-video-aggregator-web/rest/external/v2/tokens',
            video_id, 'Downloading token', headers={
                **self.geo_verification_headers(),
                'Content-Type': 'application/json; charset=utf-8',
            }, data=json.dumps({'identityToken': vrtnutoken}).encode())['vrtPlayerToken']

        video_info = self._download_json(
            f'https://media-services-public.vrt.be/media-aggregator/v2/media-items/{video_id}',
            video_id, 'Downloading video JSON', query={
                'vrtPlayerToken': vrt_player_token,
                'client': 'vrtnu-web@PROD',
            }, expected_status=400)
```